src/module/rolls/roll.ts

```diff
@@ -27,48 +52,81 @@ export type SourcedValue<T = number> = {
   value: T
 }
 
-// Input to rolling and resolution
+/**
+ * Input to rolling and resolution
+ */
 export interface PreRollOptions {
-  // Exactly one of `progress` or `action` is required
-
-  // indicates this is a progress roll, no action die will be rolled
+  // TODO: candidate for being a generic type with some variable props dependent on boolean "isProgressRoll"?
```

Comment:
In the interest of getting this merged and shipped, let's leave this as a problem for Future Rsek to solve. :)